src/compute/src/arrangement/manager.rs

```diff
@@ -170,6 +178,128 @@ impl SpecializedTraceHandle {
             }
         }
     }
+
+    /// Turns this trace into a padded version that reports empty data for all times less than the
+    /// trace's current logical compaction frontier.
+    fn into_padded(self) -> Self {
+        match self {
+            Self::RowRow(trace) => Self::RowRow(trace.into_padded()),
+        }
+    }
+}
+
+/// Handle to a trace that can be padded.
+///
+/// A padded trace contains empty data for all times greater than or equal to its `padded_since`
+/// and less than the logical compaction frontier of the inner `trace`.
+#[derive(Clone)]
+pub struct PaddedTrace<Tr> {
+    /// The wrapped trace.
+    trace: Tr,
+    /// The frontier from which the trace is padded, or `None` if it is not padded.
```

Comment:
Is there an implicit invariant that `padded_since` is less than the trace's current sice? If yes, should document and make sure we're upholding the invariant.

Reply:
Yep! Will do.

Reply:
Added the documentation and also added a check to `into_padded` to ensure the invariant also is upheld there.